### ISSUE TITLE
Possible bug with file not in a workspace. Silent failure, output log shows: TypeError: Cannot read properties of undefined (reading '0')

### ISSUE THREAD

**Issue by jameshowison** [sqlfluff/vscode-sqlfluff#75]
I installed sqlfluff via vscode.  I have sqlfluff installed via homebrew (1.4.2).

My symptom was that there was no indication in the GUI that sqlfluff ran at all.  

I tracked down some error messages in the Log (Extension Host) Output.  This shows me enabling dorzey.vscode-sqlfluff.  I then edited and then chose "Format Document".  There is no indication that anything happened in the GUI.  The log shows the error `TypeError: Cannot read properties of undefined (reading '0')`

```
2022-11-15 10:29:44.750 [info] ExtensionService#_doActivateExtension dorzey.vscode-sqlfluff, startup: false, activationEvent: 'onLanguage:sql'
2022-11-15 10:30:16.485 [error] TypeError: Cannot read properties of undefined (reading '0')
	at y1.doLint (/Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:3223)
	at /Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:3106
	at Y1.queue (/Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:737)
	at T1.<anonymous> (/Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:1630)
	at /Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:1151
	at runNextTicks (node:internal/process/task_queues:61:5)
	at listOnTimeout (node:internal/timers:528:9)
	at process.processTimers (node:internal/timers:502:7)
2022-11-15 10:30:19.444 [error] [dorzey.vscode-sqlfluff] provider FAILED
2022-11-15 10:30:19.444 [error] TypeError: Cannot read properties of undefined (reading '0')
	at j1.provideDocumentFormattingEdits (/Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:14:978)
	at B.provideDocumentFormattingEdits (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:45923)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:73880
	at ne._withAdapter (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:67325)
	at ne.$provideDocumentFormattingEdits (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:73856)
	at l._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:102:13680)
	at l._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:102:13362)
	at l._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:102:12081)
	at l._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:102:10834)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:102:8941
	at h.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:145)
	at v.deliver (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:2266)
	at g.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:1844)
	at r.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:70:16515)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:118:17104
	at h.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:145)
	at v.deliver (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:2266)
	at g.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:1844)
	at r.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:70:16515)
	at o._receiveMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:70:21327)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:70:18770
	at h.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:145)
	at v.deliver (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:2266)
	at g.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:61:1844)
	at g.acceptChunk (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:70:13346)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:70:12476
	at Socket.i (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:70:24594)
	at Socket.emit (node:events:526:28)
	at addChunk (node:internal/streams/readable:315:12)
	at readableAddChunk (node:internal/streams/readable:289:9)
	at Readable.push (node:internal/streams/readable:228:10)
	at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
```

**Comment by jameshowison** [comment]
I tried reinstalling vscode and the extension, no cigar.  Perhaps it is coming from a settings.json formatting error, but I've checked for those.

I tried setting onType and I get a similar error:

```
2022-11-16 09:14:33.453 [error] TypeError: Cannot read properties of undefined (reading '0')
	at y1.doLint (/Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:3223)
	at /Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:3106
	at Y1.queue (/Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:737)
	at T1.<anonymous> (/Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:1630)
	at /Users/jlh5498/.vscode/extensions/dorzey.vscode-sqlfluff-0.1.9/out/src/extension.js:18:1151
```

I have tried resetting all edited preferences.

Any debugging suggestions?

I understand only that extension.js is created from extension.ts but I can't investigate further than that.

**Comment by jameshowison** [comment]
Seems like this might come from "Q.workspace.workspaceFolders[0].uri.fsPath" which is at out/src/extension.js:18:3223 

Other error is coming from around extension.js:14:978 which also has similar code: "G.workspace.workspaceFolders[0].uri.fsPath)"

Perhaps this is related to this issue:
https://stackoverflow.com/questions/62354386/vscode-workspace-workspacefolders-returning-error-vscode-extensions

**Comment by jameshowison** [comment]
Ok, so opening the file in a workspace seems to have solved this error.

I'll change the name of the issue to: Possible bug with file not in a workspace.

**Comment by RobertOstermann** [comment]
@jameshowison Thanks for looking into this and narrowing down the problem. I will try to get a fix out within the next couple of days.

**Comment by RobertOstermann** [comment]
@jameshowison This should be fixed in v2.0.0. Let me know if that works for you

**Comment by jameshowison** [comment]
Yup.  Took a little bit of effort to figure out how to get back to the error (in the end I had to save my "Untitled" workspace, then find the file that created and then delete that file.  That got rid of the 'Untitled Workspace'.  Then, with 0.1.9 creating a new file and saving it as filename.sql I got a similar error to before (not identical oddly). Then I upgraded to 2.0.1 and the error went away and I saw the sqlfluff lint.

So I think this has worked!
